ui/app/AppLayouts/Profile/panels/ContactsListPanel.qml

```diff
@@ -1,57 +1,163 @@
-    signal openProfilePopup(var contact)
-    signal sendMessageActionTriggered(var contact)
-    signal openChangeNicknamePopup(var contact)
-
-    width: parent.width
-
-    model: contactList.contactsModel
-
-    delegate: ContactPanel {
-        id: panelDelegate
-        name: model.name
-        publicKey: model.pubKey
-        icon: model.icon
-        isIdenticon: model.isIdenticon
-        isContact: model.isContact
-        isBlocked: model.isBlocked
-        showSendMessageButton: contactList.showSendMessageButton
-
-        onClicked: contactList.contactClicked(model)
-        onOpenProfilePopup: contactList.openProfilePopup(model)
-        onSendMessageActionTriggered: contactList.sendMessageActionTriggered(model)
-        onOpenChangeNicknamePopup: contactList.openChangeNicknamePopup(model)
-
-        visible: {
-          if (hideBlocked && model.isBlocked) {
-            return false
-          }
-
-          return searchString === "" ||
-            panelDelegate.name.toLowerCase().includes(lowerCaseSearchString) ||
-            panelDelegate.publicKey.toLowerCase().includes(lowerCaseSearchString)
+
+    signal contactClicked(string publicKey)
+    signal openProfilePopup(string publicKey)
+    signal sendMessageActionTriggered(string publicKey)
+    signal openChangeNicknamePopup(string publicKey)
+    signal acceptContactRequest(string publicKey)
+    signal rejectContactRequest(string publicKey)
+    signal removeRejection(string publicKey)
+    signal textClicked(string publicKey)
+
+    visible: contactsList.count > 0
+
+    StyledText {
+        id: title
+        anchors.top: parent.top
+        anchors.left: parent.left
+        visible: contactListRoot.title !== ""
+        text: contactListRoot.title
+        font.weight: Font.Medium
+        font.pixelSize: 15
+        color: Style.current.secondaryText
+    }
+
+    DelegateModel {
+        id: delegateModel
+
+        function update() {
+            var visible = [];
+            for (var i = 0; i < items.count; ++i) {
+                var item = items.get(i);
+                if (panelUsage === Constants.contactsPanelUsage.verifiedMutualContacts) {
+                    if(item.model.verificationState === Constants.contactVerificationState.verified)
+                        visible.push(item);
+                }
+                else if (panelUsage === Constants.contactsPanelUsage.mutualContacts) {
+                    if(item.model.verificationState !== Constants.contactVerificationState.verified)
+                        visible.push(item);
+                }
+                else {
+                    visible.push(item);
+                }
+            }
+
+            for (i = 0; i < visible.length; ++i) {
+                item = visible[i];
+                item.inVisible = true;
+                if (item.visibleIndex !== i) {
+                    visibleItems.move(item.visibleIndex, i, 1);
+                }
+            }
+        }
```

Comment:
It's qml code smell, model filtering should not be done in an imperative way on qml/js side. We should probably use: https://github.com/oKcerG/SortFilterProxyModel/ or create our own utility. 

This comment does not require any changes for this PR, it's just to shed light on the problem. I've seen it done wrong in many places in our codebase. I will create global issue for that.

Reply:
On the other hand: why not use separate models instead of imperative filtering?
```
model: {
    if (panelUsage === Constants.contactsPanelUsage.verifiedMutualContacts) {
        return verifiedMutualContactsModel
    }
    if (panelUsage === Constants.contactsPanelUsage.mutualContacts) {
        return mutualContactsModel
    }
    return contactsModel
}
```
These models are exposed in `profile_section/constacts/view.nim` already.

Reply:
@osmaczko yes that's true, and I would say nothing new since there are more places which should be improved in the same way, hope we will handle that properly in the c++ app having proxy model on the c++ side where is needed.

> why not use separate models instead of imperative filtering?

I could, but `verifiedMutualContacts ` is just a subset of `mutualContacts ` and decided to do it this way.